academy/0.0-B9lab-Blockchains/4_consensus.md

```diff
@@ -0,0 +1,112 @@
+
+## Delegated-Proof-of-Stake (DPoS)
+
+An extension of proof-of-stake algorithms is called delegated-proof-of-stake (DPoS). It is used for example in BitShares, EOS, Steem, Lisk and Tron.
+
+<HighlightBox type="info">
+    [Bitshares](https://bitshares.org/) is an open-source, blockchain-based financial platform launched in 2014 and the name of the corresponding cryptocurrency. It provides a decentralised asset exchange – comparable to a stock exchange like the New York Stock Exchange (NYSE) but focused on cryptocurrencies.
+</HighlightBox>
+
+The algorithm is called DPoS because as in PoS the value of a vote is determined by the stake, i.e. tokens held by a user. However, in DPoS systems, there is a fixed validator set. For example, in EOS, there are only 21 validators that participate in consensus. In pure PoS systems, there is no fixed validator set and the number of potential validators that can participate in consensus is dependent on the total supply of tokens in circulation.
+
+In this type of consensus mechanism, so-called **"witnesses"** are elected by the stakeholders of the network. Witnesses secure the network. Afterwards, several witnesses are chosen for the block creation so that they represent at least 50% of the stakeholders' votes.
+
+![DPoS](images/00_07_delegated_pos_dark.png)
+
+Witnesses are paid for their services: They are paid fees for creating and validating blocks. This economic incentive to become a witness also leads to competition potentially increasing with each new member because the number of witnesses is limited.
+
+In case a witness misbehaves, the network's community is able to withdraw their votes for a single witness, i.e. fire the witness. Witnesses that no longer hold enough votes lose their income basis.
+
+Alongside ascribing the role of witnesses to some participants, DPoS networks also elect **"delegates"**. Delegates are a group of participants that supervise network governance and performance, and propose changes that are then voted on by the entire network.
+
+Many consider DPoS algorithms superior to PoW and PoS because of their fast block creation, a high degree of security, energy efficiency, level of integrity, and democratic structure. However, DPoS systems are less decentralized than PoW and PoS systems because they have fixed validator sets and higher barriers to entry.
+
+## Practical Byzantine Fault Tolerance (PBFT)
+
+PBFT was published in 1999 and arose from academia. It is a three-phase protocol, in which the client sends a request to a so-called primary. 
+
+In the first phase, the primary broadcasts the request with a sequence number to the replicas. Then the replicas agree on the sequence number and create a message. 
```

Comment:
What is a replica? I would introduce a sentence or expand the sentence to explain very briefly the concept.